examples/pigweed-app/esp32/main/main.cpp

```diff
@@ -26,31 +26,44 @@
 #include "pw_sys_io/sys_io.h"
 #include "pw_sys_io_esp32/init.h"
 
+#include "PigweedLoggerMutex.h"
 #include "RpcService.h"
+#include <support/logging/CHIPLogging.h>
 
 #include "lwip/err.h"
 #include "lwip/sockets.h"
 #include "lwip/sys.h"
 
 const char * TAG = "chip-pigweed-app";
 
+bool uartInitialised = false;
+// ESP32 Logging
+
+extern "C" void __wrap_esp_log_write(esp_log_level_t level, const char * tag, const char * format, ...)
+{
+    va_list v;
+    va_start(v, format);
+#if CONFIG_LOG_DEFAULT_LEVEL > 1
+    if (uartInitialised == true)
+    {
+        char formattedMsg[CHIP_CONFIG_LOG_MESSAGE_MAX_SIZE];
+        size_t len = vsnprintf(formattedMsg, sizeof formattedMsg, format, v);
+        if (len >= sizeof formattedMsg)
+        {
+            formattedMsg[sizeof formattedMsg - 1] = '\0';
```

Comment:
vsnprintf manpage: If the output was truncated due to this limit, then the return value is the number of characters (excluding the  terminating  null  byte)  which  would have been written to the final string if enough space had been available.

So, in this check you should ideally set the len to be `sizeof formattedMsg - 1` ? Else, it will pass the incorrect (higher) len to the putString function

Reply:
Yes, added the fix.